be/src/runtime/mem_tracker.h

```diff
@@ -164,11 +167,16 @@ class MemTracker : public std::enable_shared_from_this<MemTracker> {
     /// other callers that may not tolerate allocation failures have a better chance
     /// of success. Returns true if the consumption was successfully updated.
     WARN_UNUSED_RESULT
-    bool TryConsume(int64_t bytes, MemLimit mode = MemLimit::HARD) {
+    Status TryConsume(int64_t bytes, MemLimit mode = MemLimit::HARD) {
         // DCHECK_GE(bytes, 0);
         if (bytes <= 0) {
             Release(-bytes);
-            return true;
+            return Status::OK();
+        }
+        if (MemInfo::current_mem() + bytes >= MemInfo::mem_limit()) {
+            return Status::MemoryLimitExceeded(fmt::format(
+                    "{}: TryConsume failed, bytes={} process whole consumption={}  mem limit={}",
+                    label_, bytes, MemInfo::current_mem(), MemInfo::mem_limit()));
```

Comment:
Printing MemTracker::MemLimitExceeded has a certain cost, confirm that it will not be printed frequently.

Reply:
Status::MemoryLimitExceeded  not MemTracker::MemLimitExceeded